basheetka.bb

```diff
@@ -100,17 +100,27 @@
   (let [{:keys [name result]} (tasks/current-task)]
     (swap! state assoc name result)))
 
-;;; Create initial bb.edn
-
-(defn init [_]
-  (println "Initialising bb.edn...")
-  (bb-edn-write initial-bb-edn))
+;;; Create initial bb.edn (added overwrite existing with --force option)
+
+(defn init [{:keys [opts]}]
+  (let [bb-exists? (.exists (io/file *bb-edn-file*))
+        force? (:force opts)]
+    (when (and (not force?) bb-exists?)
+      (print "bb.edn file already exists. Do you want to overwrite? [y/n] ")
+      (flush)
+      (let [response (read-line)]
+        (when (not= (str/lower-case response) "y")
+          (println "Using existing bb.edn")        
+          (System/exit 0))))
+  
+    (if bb-exists? (println "Overwriting bb.edn") (println "Initializing bb.edn"))
+    (bb-edn-write initial-bb-edn)))
 
 (defn help [_]
-  (println  "Usage: ./basheetka.bb init"))
+  (println  "Usage: ./basheetka.bb init [optional] --force"))
```

Comment:
Minor thought. Do you think it looks more idiomatic to put the option itself in square brackets? Something like:

`Usage: ./basheetka.bb init [--force]`

One day we can use the [built in facility to show help](https://github.com/babashka/cli#help) but that's probably overkill right now.

Reply:
I thought the same before, I was also not sure how to represent optional sub-command in usage. 

Similarly, strings like "Initializing bb.edn..." changed into "Initializing bb.edn", etc., it's upto you, how you want them to be, I didn't thought about them much.

So, whatever feels right to you, will be much more idiomatic, because I have no idea about conventions to use for defining such strings.

Reply:
> Similarly, strings like "Initializing bb.edn..." changed into "Initializing bb.edn", etc., it's upto you, how you want them to be, I didn't thought about them much.

I think your removal of "..." makes this better :+1:

> So, whatever feels right to you, will be much more idiomatic, because I have no idea about conventions to use for defining such strings.

Let's go with `Usage: ./basheetka.bb init [--force]` if you don't mind, then at some stage in the future, we'll probably reconsider anyway.

Reply:
The only other thing to consider is updating the tests to cover the overwriting (or not) of `bb.edn`. Probably a good idea seeing an error might erase someone's `bb.edn`. If you look in `basheetka_test.bb` you'll see:

```clojure
(deftest round-trip-test
  ...)
```

This might be the right place to consider.

Reply:
Updated tests for overwrite case too.

I am totally beginner in writing test cases. I am not sure about wether I wrote them correctly or not, specifically about comparing `bb.edn` contents in each test.

Reply:
Great job with the tests!

At some stage, I can see us pulling out the `round-trip-test` into testing the `bb.edn` creation/overwrite functionality separate to the csv file functionality, but we're good for now.

I committed a bit of a tidy up. Feel free to take a look. I like the use of `with-out-str` ... though I think you can use `{:out :string}` as a convenience. Will take a look at that now.

Otherwise, I'll merge in the pull request shortly :)